Composer/packages/server/src/models/bot/botProject.ts

```diff
@@ -406,10 +406,10 @@ export class BotProject {
     if (index === -1) {
       throw new Error(`no such file at ${relativePath}`);
     }
+    this.files.splice(index, 1);
 
     const absolutePath = `${this.dir}/${relativePath}`;
     await this.fileStorage.removeFile(absolutePath);
-    this.files.splice(index, 1);
```

Comment:
`this.files` can be different after `await` when concurrency requests come in. this is the bug code cause `unexpected file delete/not-delete`